hack/observability/kube-state-metrics/kustomization.yaml

```diff
@@ -0,0 +1,12 @@
+resources:
+  - ../namespace.yaml
+  # The kube-state-metrics chart needs to be referenced in a sub-kustomization.yaml,
```

Comment:
How does this work exactly? Are we overwriting a ConfigMap from the chart?

Reply:
Nope.

The ConfigMapGenerator below creates a configmap with name `kube-state-metrics-crd-config` which results in a configmap called `kube-state-metrics-crd-config-<suffix>`.

The helm chart has references to a configmap named `kube-state-metrics-crd-config` (which is the same as our configmapgenerator) (it does not have the configmap, just the volume mount reference inside the deployment).

Kustomize then replaces the referenced name in the deployments volume reference from `kube-state-metrics-crd-config` to `kube-state-metrics-crd-config-<suffix>`, where suffix is deterministic from the content of the configmap.

This functionality would for every yaml referenced by an entry in `resources`, but does not work if we would move the content of `hack/observability/kube-state-metrics/chart/kustomization.yaml` to here.

Reply:
Ah perfect thx.

Please consider expanding the current comment. I think without background knowledge it's hard to infer that based on the current comment.

Reply:
Perfect, thx!